internal/launcher/launcher.go

```diff
@@ -0,0 +1,158 @@
+// Package launcher is the entry point for the k6 command.
+package launcher
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/grafana/k6deps"
+	"go.k6.io/k6/cmd/state"
+	"go.k6.io/k6/internal/build"
+	k6Cmd "go.k6.io/k6/internal/cmd"
+)
+
+// Execute runs the k6 command.
+func Execute() {
+	gs := state.NewGlobalState(context.Background())
+
+	newLauncher(gs).launch()
+}
+
+// launcher is a k6 launcher
+type launcher struct {
+	gs *state.GlobalState
+	// function to fall back if binary provisioning is not required
+	fallback func(gs *state.GlobalState)
```

Comment:
nit; I'm not really convinced that `fallback` is a good name for this, as I'd expect not to have to provision the binary the "happy" and most common path.